thirdparty/cmdconfig/commands/cmdeval/cmdeval.go

```diff
@@ -211,7 +153,33 @@ data: {}
 	if err != nil {
 		return nil, err
 	}
-	return []*yaml.RNode{rc}, nil
+	return rc, nil
+}
+
+func (r *EvalFnRunner) getFunctionSpec() (*runtimeutil.FunctionSpec, []string, error) {
+	fn := &runtimeutil.FunctionSpec{}
+	var execArgs []string
+	if r.Image != "" {
+		fn.Container.Image = r.Image
+	} else if r.Exec != "" {
+		// check the flags that doesn't make sense with exec function
+		// --mount, --as-current-user, --network and --env are
+		// only used with container functions
+		if r.AsCurrentUser || r.Network ||
+			len(r.Mounts) != 0 || len(r.Env) != 0 {
+			return nil, nil, fmt.Errorf("--mount, --as-current-user, --network and --env can only be used with container functions")
+		}
+		if !strings.Contains(r.Exec, " ") {
+			fn.Exec.Path = r.Exec
+		} else {
+			// we use sh to run the commands so we don't need to
```

Comment:
relying on sh is an additional external dependency and extra process. https://pkg.go.dev/github.com/google/shlex may be a good tool to parse. 

```
shlex.Split("one \"two three\" four") -> []string{"one", "two three", "four"}
```

Reply:
That's cool! I will try it.

Reply:
just got another though - on another hand... your approach supports pipes, storing to files and all other 'sh' things.. maybe it's better...

Reply:
> just got another thought - on another hand: your approach supports pipes, storing to files and all other 'sh' things.. maybe it's better...

I don't think supporting those features is the goal here.